# Lab 2

In [9]:
import warnings
warnings.simplefilter('ignore')

import pandas as pd

df = pd.read_csv('s3://ia340suzuki/Crime_Data_from_2020_to_Present.csv') # load the data from s3
df.head(10) #show the top 10 rows

DR_NO               Date Rptd                DATE OCC  TIME OCC  AREA  \
0  211507896  04/11/2021 12:00:00 AM  11/07/2020 12:00:00 AM       845    15   
1  201516622  10/21/2020 12:00:00 AM  10/18/2020 12:00:00 AM      1845    15   
2  240913563  12/10/2024 12:00:00 AM  10/30/2020 12:00:00 AM      1240     9   
3  210704711  12/24/2020 12:00:00 AM  12/24/2020 12:00:00 AM      1310     7   
4  201418201  10/03/2020 12:00:00 AM  09/29/2020 12:00:00 AM      1830    14   
5  240412063  12/11/2024 12:00:00 AM  11/11/2020 12:00:00 AM      1210     4   
6  240317069  12/16/2024 12:00:00 AM  04/16/2020 12:00:00 AM      1350     3   
7  201115217  10/29/2020 12:00:00 AM  07/07/2020 12:00:00 AM      1400    11   
8  241708596  04/20/2024 12:00:00 AM  03/02/2020 12:00:00 AM      1200    17   
9  242113813  12/18/2024 12:00:00 AM  09/01/2020 12:00:00 AM       900    21   

     AREA NAME  Rpt Dist No  Part 1-2  Crm Cd  \
0  N Hollywood         1502         2     354   
1  N Hollywood         1521         1     230   
2     Van Nuys          933         2     354   
3     Wilshire          782         1     331   
4      Pacific         1454         1     420   
5   Hollenbeck          429         2     354   
6    Southwest          396         2     354   
7    Northeast         1133         2     812   
8   Devonshire         1729         2     354   
9      Topanga         2196         2     354   

                                         Crm Cd Desc  ... Status  Status Desc  \
0                                  THEFT OF IDENTITY  ...     IC  Invest Cont   
1     ASSAULT WITH DEADLY WEAPON, AGGRAVATED ASSAULT  ...     IC  Invest Cont   
2                                  THEFT OF IDENTITY  ...     IC  Invest Cont   
3  THEFT FROM MOTOR VEHICLE - GRAND ($950.01 AND ...  ...     IC  Invest Cont   
4    THEFT FROM MOTOR VEHICLE - PETTY ($950 & UNDER)  ...     IC  Invest Cont   
5                                  THEFT OF IDENTITY  ...     IC  Invest Cont   
6                                  THEFT OF IDENTITY  ...     IC  Invest Cont   
7  CRM AGNST CHLD (13 OR UNDER) (14-15 & SUSP 10 ...  ...     AO  Adult Other   
8                                  THEFT OF IDENTITY  ...     IC  Invest Cont   
9                                  THEFT OF IDENTITY  ...     IC  Invest Cont   

  Crm Cd 1 Crm Cd 2  Crm Cd 3 Crm Cd 4  \
0    354.0      NaN       NaN      NaN   
1    230.0      NaN       NaN      NaN   
2    354.0      NaN       NaN      NaN   
3    331.0      NaN       NaN      NaN   
4    420.0      NaN       NaN      NaN   
5    354.0      NaN       NaN      NaN   
6    354.0      NaN       NaN      NaN   
7    812.0    860.0       NaN      NaN   
8    354.0      NaN       NaN      NaN   
9    354.0      NaN       NaN      NaN   

                                   LOCATION Cross Street      LAT       LON  
0   7800    BEEMAN                       AV          NaN  34.2124 -118.4092  
1           ATOLL                        AV     N  GAULT  34.1993 -118.4203  
2  14600    SYLVAN                       ST          NaN  34.1847 -118.4509  
3   6000    COMEY                        AV          NaN  34.0339 -118.3747  
4                   4700    LA VILLA MARINA          NaN  33.9813 -118.4350  
5   5300    CRONUS                       ST          NaN  34.0830 -118.1678  
6    900 W  40TH                         PL          NaN  34.0100 -118.2900  
7   3000    ACRESITE                     ST          NaN  34.1107 -118.2589  
8  17700    SIMONDS                      ST          NaN  34.2763 -118.5210  
9  20900    MARMORA                      ST          NaN  34.1493 -118.5886  

[10 rows x 28 columns]

In [8]:
df.shape #show the number of rows and columns 

(1004991, 28)

## My data has 1,004,991 rows and 28 columns. Rows are text and numerical

In [14]:
df.sort_values(by='LOCATION')

DR_NO               Date Rptd                DATE OCC  TIME OCC  \
514080  221421172  12/17/2022 12:00:00 AM  12/04/2022 12:00:00 AM      1200   
559191  221416993  09/27/2022 12:00:00 AM  08/20/2022 12:00:00 AM      1500   
432903  221409015  04/09/2022 12:00:00 AM  04/09/2022 12:00:00 AM      1930   
17194   201420046  11/09/2020 12:00:00 AM  11/08/2020 12:00:00 AM      1930   
617129  221414581  08/03/2022 12:00:00 AM  08/03/2022 12:00:00 AM       310   
...           ...                     ...                     ...       ...   
793541  231105669  02/13/2023 12:00:00 AM  02/13/2023 12:00:00 AM      1620   
653345  231112618  08/24/2023 12:00:00 AM  08/18/2023 12:00:00 AM      1430   
163453  201112781  08/22/2020 12:00:00 AM  08/22/2020 12:00:00 AM      1715   
134615  201109722  05/25/2020 12:00:00 AM  05/24/2020 12:00:00 AM       745   
597084  221109208  05/15/2022 12:00:00 AM  05/13/2022 12:00:00 AM      1430   

        AREA  AREA NAME  Rpt Dist No  Part 1-2  Crm Cd  \
514080    14    Pacific         1431         1     341   
559191    14    Pacific         1431         1     440   
432903    14    Pacific         1431         2     740   
17194     14    Pacific         1431         1     310   
617129    14    Pacific         1431         1     420   
...      ...        ...          ...       ...     ...   
793541    11  Northeast         1101         1     230   
653345    11  Northeast         1101         1     510   
163453    11  Northeast         1101         1     330   
134615    11  Northeast         1101         1     330   
597084    11  Northeast         1101         1     330   

                                              Crm Cd Desc  ... Status  \
514080  THEFT-GRAND ($950.01 & OVER)EXCPT,GUNS,FOWL,LI...  ...     IC   
559191                 THEFT PLAIN - PETTY ($950 & UNDER)  ...     IC   
432903  VANDALISM - FELONY ($400 & OVER, ALL CHURCH VA...  ...     IC   
17194                                            BURGLARY  ...     IC   
617129    THEFT FROM MOTOR VEHICLE - PETTY ($950 & UNDER)  ...     IC   
...                                                   ...  ...    ...   
793541     ASSAULT WITH DEADLY WEAPON, AGGRAVATED ASSAULT  ...     IC   
653345                                   VEHICLE - STOLEN  ...     IC   
163453                              BURGLARY FROM VEHICLE  ...     IC   
134615                              BURGLARY FROM VEHICLE  ...     IC   
597084                              BURGLARY FROM VEHICLE  ...     IC   

        Status Desc Crm Cd 1 Crm Cd 2  Crm Cd 3 Crm Cd 4  \
514080  Invest Cont    341.0      NaN       NaN      NaN   
559191  Invest Cont    440.0      NaN       NaN      NaN   
432903  Invest Cont    740.0      NaN       NaN      NaN   
17194   Invest Cont    310.0      NaN       NaN      NaN   
617129  Invest Cont    420.0      NaN       NaN      NaN   
...             ...      ...      ...       ...      ...   
793541  Invest Cont    230.0      NaN       NaN      NaN   
653345  Invest Cont    510.0      NaN       NaN      NaN   
163453  Invest Cont    330.0      NaN       NaN      NaN   
134615  Invest Cont    330.0      NaN       NaN      NaN   
597084  Invest Cont    330.0      NaN       NaN      NaN   

                                     LOCATION  \
514080                             00    17TH   
559191  00    17TH                         AV   
432903  00    17TH                         AV   
17194   00    17TH                         AV   
617129  00    17TH                         AV   
...                                       ...   
793541        ZOO                          DR   
653345        ZOO                          DR   
163453        ZOO                          DR   
134615        ZOO                          DR   
597084        ZOO                          DR   

                           Cross Street      LAT       LON  
514080                              NaN  33.9864 -118.4724  
559191                              NaN  33.9867 -118.472

In [15]:
#rename columns
df = df.rename(columns={"AREA NAME": "REGION"}) 
df.head()

DR_NO               Date Rptd                DATE OCC  TIME OCC  AREA  \
0  211507896  04/11/2021 12:00:00 AM  11/07/2020 12:00:00 AM       845    15   
1  201516622  10/21/2020 12:00:00 AM  10/18/2020 12:00:00 AM      1845    15   
2  240913563  12/10/2024 12:00:00 AM  10/30/2020 12:00:00 AM      1240     9   
3  210704711  12/24/2020 12:00:00 AM  12/24/2020 12:00:00 AM      1310     7   
4  201418201  10/03/2020 12:00:00 AM  09/29/2020 12:00:00 AM      1830    14   

        REGION  Rpt Dist No  Part 1-2  Crm Cd  \
0  N Hollywood         1502         2     354   
1  N Hollywood         1521         1     230   
2     Van Nuys          933         2     354   
3     Wilshire          782         1     331   
4      Pacific         1454         1     420   

                                         Crm Cd Desc  ... Status  Status Desc  \
0                                  THEFT OF IDENTITY  ...     IC  Invest Cont   
1     ASSAULT WITH DEADLY WEAPON, AGGRAVATED ASSAULT  ...     IC  Invest Cont   
2                                  THEFT OF IDENTITY  ...     IC  Invest Cont   
3  THEFT FROM MOTOR VEHICLE - GRAND ($950.01 AND ...  ...     IC  Invest Cont   
4    THEFT FROM MOTOR VEHICLE - PETTY ($950 & UNDER)  ...     IC  Invest Cont   

  Crm Cd 1 Crm Cd 2  Crm Cd 3 Crm Cd 4  \
0    354.0      NaN       NaN      NaN   
1    230.0      NaN       NaN      NaN   
2    354.0      NaN       NaN      NaN   
3    331.0      NaN       NaN      NaN   
4    420.0      NaN       NaN      NaN   

                                   LOCATION Cross Street      LAT       LON  
0   7800    BEEMAN                       AV          NaN  34.2124 -118.4092  
1           ATOLL                        AV     N  GAULT  34.1993 -118.4203  
2  14600    SYLVAN                       ST          NaN  34.1847 -118.4509  
3   6000    COMEY                        AV          NaN  34.0339 -118.3747  
4                   4700    LA VILLA MARINA          NaN  33.9813 -118.4350  

[5 rows x 28 columns]

### Renamed AREA NAME to REGION

In [16]:
df.isnull().sum() #show the number of rows containing null value

DR_NO                   0
Date Rptd               0
DATE OCC                0
TIME OCC                0
AREA                    0
REGION                  0
Rpt Dist No             0
Part 1-2                0
Crm Cd                  0
Crm Cd Desc             0
Mocodes            151619
Vict Age                0
Vict Sex           144644
Vict Descent       144656
Premis Cd              16
Premis Desc           588
Weapon Used Cd     677744
Weapon Desc        677744
Status                  1
Status Desc             0
Crm Cd 1               11
Crm Cd 2           935831
Crm Cd 3          1002677
Crm Cd 4          1004927
LOCATION                0
Cross Street       850755
LAT                     0
LON                     0
dtype: int64

In [20]:
df.dropna() #drop null values


DR_NO               Date Rptd                DATE OCC  TIME OCC  \
27751   201904032  01/02/2020 12:00:00 AM  01/01/2020 12:00:00 AM      2135   
134755  200613424  08/02/2020 12:00:00 AM  08/02/2020 12:00:00 AM      2030   
363765  210617136  10/08/2021 12:00:00 AM  10/07/2021 12:00:00 AM      1950   
372524  210209196  05/08/2021 12:00:00 AM  05/08/2021 12:00:00 AM       230   
490101  220600626  04/27/2022 12:00:00 AM  04/23/2022 12:00:00 AM      2300   
537834  221718232  12/25/2022 12:00:00 AM  12/25/2022 12:00:00 AM      1150   
586003  221401314  11/10/2022 12:00:00 AM  11/10/2022 12:00:00 AM      2117   
728576  231717599  11/15/2023 12:00:00 AM  11/15/2023 12:00:00 AM       400   
809419  231915572  10/21/2023 12:00:00 AM  10/21/2023 12:00:00 AM         1   
934749  241905348  02/04/2024 12:00:00 AM  02/03/2024 12:00:00 AM      1100   

        AREA      REGION  Rpt Dist No  Part 1-2  Crm Cd  \
27751     19     Mission         1924         1     761   
134755     6   Hollywood          657         1     761   
363765     6   Hollywood          659         1     121   
372524     2     Rampart          279         1     210   
490101     6   Hollywood          646         1     821   
537834    17  Devonshire         1797         1     122   
586003    14     Pacific         1452         2     910   
728576    17  Devonshire         1738         1     210   
809419    19     Mission         1902         1     210   
934749    19     Mission         1983         1     820   

                                              Crm Cd Desc  ... Status  \
27751                                     BRANDISH WEAPON  ...     AA   
134755                                    BRANDISH WEAPON  ...     AO   
363765                                     RAPE, FORCIBLE  ...     IC   
372524                                            ROBBERY  ...     AO   
490101  SODOMY/SEXUAL CONTACT B/W PENIS OF ONE PERS TO...  ...     IC   
537834                                    RAPE, ATTEMPTED  ...     AA   
586003                                         KIDNAPPING  ...     IC   
728576                                            ROBBERY  ...     IC   
809419                                            ROBBERY  ...     AA   
934749                                    ORAL COPULATION  ...     AO   

         Status Desc Crm Cd 1 Crm Cd 2  Crm Cd 3 Crm Cd 4  \
27751   Adult Arrest    761.0    930.0     997.0    998.0   
134755   Adult Other    761.0    920.0     930.0    998.0   
363765   Invest Cont    121.0    210.0     910.0    998.0   
372524   Adult Other    210.0    510.0     910.0    998.0   
490101   Invest Cont    230.0    821.0     910.0    998.0   
537834  Adult Arrest    122.0    230.0     910.0    998.0   
586003   Invest Cont    812.0    860.0     910.0    998.0   
728576   Invest Cont    210.0    230.0     761.0    998.0   
809419  Adult Arrest    210.0    250.0     761.0    998.0   
934749   Adult Other    761.0    820.0     910.0    998.0   

                               LOCATION                     Cross Street  \
27751   ASTORIA                      ST  SAN FERNANDO                 RD   
134755                          WESTERN                          ROMAINE   
363765                        NORMANDIE                       DE LONGPRE   
372524                     JAMES M WOOD                            GREEN   
490101                            SELMA                       LAS PALMAS   
537834  PARTHENIA                    ST                      HAYVENHURST   
586003                       WASHINGTON                         SPEEDWAY   
728576  HASKELL                      AV  SAN FERNANDO                 BL   
809419                             POLK                           BORDEN   
934749                           BURNET                        PARTHENIA   

            LAT       LON  
27751   34.2949 -118.4571  
134755  34.0885 -118.3092  
363765  34.0966 -118.3005  
372524  34.0503 -118.2720  
490101  34.0997 -118.3363  
537834  34.

#### Using dropna(), I now have 10 rows and 28 columns. Before, I had over 1 million rows.

In [21]:
df.isnull().sum()

DR_NO             0
Date Rptd         0
DATE OCC          0
TIME OCC          0
AREA              0
REGION            0
Rpt Dist No       0
Part 1-2          0
Crm Cd            0
Crm Cd Desc       0
Mocodes           0
Vict Age          0
Vict Sex          0
Vict Descent      0
Premis Cd         0
Premis Desc       0
Weapon Used Cd    0
Weapon Desc       0
Status            0
Status Desc       0
Crm Cd 1          0
Crm Cd 2          0
Crm Cd 3          0
Crm Cd 4          0
LOCATION          0
Cross Street      0
LAT               0
LON               0
dtype: int64

In [ ]:
df_clean = df.dropna()

In [28]:
df[(df['LAT']>= 34.1)]


DR_NO               Date Rptd                DATE OCC  TIME OCC  \
27751   201904032  01/02/2020 12:00:00 AM  01/01/2020 12:00:00 AM      2135   
537834  221718232  12/25/2022 12:00:00 AM  12/25/2022 12:00:00 AM      1150   
728576  231717599  11/15/2023 12:00:00 AM  11/15/2023 12:00:00 AM       400   
809419  231915572  10/21/2023 12:00:00 AM  10/21/2023 12:00:00 AM         1   
934749  241905348  02/04/2024 12:00:00 AM  02/03/2024 12:00:00 AM      1100   

        AREA      REGION  Rpt Dist No  Part 1-2  Crm Cd      Crm Cd Desc  ...  \
27751     19     Mission         1924         1     761  BRANDISH WEAPON  ...   
537834    17  Devonshire         1797         1     122  RAPE, ATTEMPTED  ...   
728576    17  Devonshire         1738         1     210          ROBBERY  ...   
809419    19     Mission         1902         1     210          ROBBERY  ...   
934749    19     Mission         1983         1     820  ORAL COPULATION  ...   

       Status   Status Desc Crm Cd 1 Crm Cd 2  Crm Cd 3 Crm Cd 4  \
27751      AA  Adult Arrest    761.0    930.0     997.0    998.0   
537834     AA  Adult Arrest    122.0    230.0     910.0    998.0   
728576     IC   Invest Cont    210.0    230.0     761.0    998.0   
809419     AA  Adult Arrest    210.0    250.0     761.0    998.0   
934749     AO   Adult Other    761.0    820.0     910.0    998.0   

                               LOCATION                     Cross Street  \
27751   ASTORIA                      ST  SAN FERNANDO                 RD   
537834  PARTHENIA                    ST                      HAYVENHURST   
728576  HASKELL                      AV  SAN FERNANDO                 BL   
809419                             POLK                           BORDEN   
934749                           BURNET                        PARTHENIA   

            LAT       LON  
27751   34.2949 -118.4571  
537834  34.2285 -118.4939  
728576  34.2692 -118.4789  
809419  34.3103 -118.4467  
934749  34.2282 -118.4633  

[5 rows x 28 columns]

In [30]:
df['AREA'] #view AREA column

27751     19
134755     6
363765     6
372524     2
490101     6
537834    17
586003    14
728576    17
809419    19
934749    19
Name: AREA, dtype: int64

# Filter data to show Latitude greater than 34.1 degrees. I also sorted dtat to only show AREA.

In [29]:
df[0:5]

DR_NO               Date Rptd                DATE OCC  TIME OCC  \
27751   201904032  01/02/2020 12:00:00 AM  01/01/2020 12:00:00 AM      2135   
134755  200613424  08/02/2020 12:00:00 AM  08/02/2020 12:00:00 AM      2030   
363765  210617136  10/08/2021 12:00:00 AM  10/07/2021 12:00:00 AM      1950   
372524  210209196  05/08/2021 12:00:00 AM  05/08/2021 12:00:00 AM       230   
490101  220600626  04/27/2022 12:00:00 AM  04/23/2022 12:00:00 AM      2300   

        AREA     REGION  Rpt Dist No  Part 1-2  Crm Cd  \
27751     19    Mission         1924         1     761   
134755     6  Hollywood          657         1     761   
363765     6  Hollywood          659         1     121   
372524     2    Rampart          279         1     210   
490101     6  Hollywood          646         1     821   

                                              Crm Cd Desc  ... Status  \
27751                                     BRANDISH WEAPON  ...     AA   
134755                                    BRANDISH WEAPON  ...     AO   
363765                                     RAPE, FORCIBLE  ...     IC   
372524                                            ROBBERY  ...     AO   
490101  SODOMY/SEXUAL CONTACT B/W PENIS OF ONE PERS TO...  ...     IC   

         Status Desc Crm Cd 1 Crm Cd 2  Crm Cd 3 Crm Cd 4  \
27751   Adult Arrest    761.0    930.0     997.0    998.0   
134755   Adult Other    761.0    920.0     930.0    998.0   
363765   Invest Cont    121.0    210.0     910.0    998.0   
372524   Adult Other    210.0    510.0     910.0    998.0   
490101   Invest Cont    230.0    821.0     910.0    998.0   

                               LOCATION                     Cross Street  \
27751   ASTORIA                      ST  SAN FERNANDO                 RD   
134755                          WESTERN                          ROMAINE   
363765                        NORMANDIE                       DE LONGPRE   
372524                     JAMES M WOOD                            GREEN   
490101                            SELMA                       LAS PALMAS   

            LAT       LON  
27751   34.2949 -118.4571  
134755  34.0885 -118.3092  
363765  34.0966 -118.3005  
372524  34.0503 -118.2720  
490101  34.0997 -118.3363  

[5 rows x 28 columns]